### DISCUSSION TITLE
Request is fired on mount even when key is the same

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#1066] [Q&A]
Is this the standard behaviour that all swr hooks refetch the data on mount even if the key hasn't changed? What is the reasoning for that? I have hook that downloads large files and I have the files checksum as one of the cache key and even when the checksum stays the same between mounts/rerenders, the fetch method of the swr hook is still called. 

When I set "revalidateOnMount" to false then the fetch function is **never** called.

Where is my misunderstanding?

**Comment by ErnestBrandi** [comment]
Hi, you don't say _what behaviour_ you would prefer here.

**Reply by pke** [reply]
That it is not refetched if the key hasn't changed
